### PR TITLE
Fix serialization of `search_analyzer`.

### PR DESCRIPTION
**Pull request by jpountz**
We currently have a bug that it will be omitted if the index analyzer is the
default analyzer.

### CODE REVIEW

**Comment by brwe** [comment]
Does that fix https://github.com/elastic/elasticsearch/issues/16239 ?

**Comment by jpountz** [comment]
Not entirely, there are other things to fix to completely fix #16239 that I still need to figure out.

**Comment by rjernst** [comment]
Can you add tests for each part of the boolean clauses? Seems like there are still cases missed by tests (eg this underscore check?).

LGTM

**Comment by jpountz** [comment]
@rjernst I pushed an alternative that doesn't require that we check for analyzer names that start with an underscore.

File: core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java [hunk 1]
```diff
@@ -439,15 +437,26 @@ protected void doXContentBody(XContentBuilder builder, boolean includeDefaults,
...
+        } else {
+            boolean hasDefaultIndexAnalyzer = fieldType().indexAnalyzer().name().equals("default");
+            boolean hasDifferentSearchAnalyzer = fieldType().searchAnalyzer().name().equals(fieldType().indexAnalyzer().name()) == false;
+            boolean hasDifferentSearchQuoteAnalyzer = fieldType().searchAnalyzer().name().equals(fieldType().searchQuoteAnalyzer().name()) == false;
+            if (includeDefaults || hasDefaultIndexAnalyzer == false || hasDifferentSearchAnalyzer || hasDifferentSearchQuoteAnalyzer) {
+                builder.field("analyzer", fieldType().indexAnalyzer().name());
+                if (hasDifferentSearchAnalyzer) {
+                    builder.field("search_analyzer", fieldType().searchAnalyzer().name());
+                }
+                if (hasDifferentSearchQuoteAnalyzer) {
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java]: Should this be inside the search analyzer block above? I thought we required search_analyzer was set of specifying search_quote_analyzer?

**Comment by rjernst** [comment]
Much better, thank you for the simplification! LGTM

**Pull request merged**